src/libs/dialog/src/legacy_dialog.cpp

```diff
@@ -0,0 +1,648 @@
+
+    // Divider
+    {
+        sprite_data = SPRITE_DATA[9];
+        const auto offset = static_cast<float>(formattedLinks_.size()) * static_cast<float>(lineHeight_) / vScale;
+        sprite_data.position.top -= offset;
+        sprite_data.position.bottom -= offset;
+
+        const auto &vertices = createSpriteMesh(sprite_data);
+        pV[vi++] = vertices[0];
+        pV[vi++] = vertices[1];
+        pV[vi++] = vertices[2];
+        pV[vi++] = vertices[3];
+    }
+
+    RenderService->UnLockVertexBuffer(backVertexBuffer_);
+
+    backNeedsUpdate_ = false;
+}
+
+void LegacyDialog::DrawBackground(size_t start, size_t count)
+{
+    RenderService->TextureSet(0, interfaceTexture_);
+    RenderService->DrawBuffer(backVertexBuffer_, sizeof(XI_TEX_VERTEX), backIndexBuffer_, 0, SPRITE_COUNT * 4,
+                              start * 6, count * 2, "texturedialogfon");
+}
+
+void LegacyDialog::UpdateHeadModel(const std::string &headModelPath)
+{
+    core.EraseEntity(headModel_);
+
+    const std::string newHeadModelPath = fmt::format("Heads/{}", headModelPath);
+
+    if (headModelPath_ != newHeadModelPath)
+    {
+        headModelPath_ = newHeadModelPath;
+        headModel_ = core.CreateEntity("MODELR");
+        auto gs = static_cast<VGEOMETRY *>(core.GetService("geometry"));
+        gs->SetTexturePath("characters\\");
+
+        core.Send_Message(headModel_, "ls", MSG_MODEL_LOAD_GEO, headModelPath_.c_str());
+        core.Send_Message(headModel_, "ls", MSG_MODEL_LOAD_ANI, headModelPath_.c_str());
+
+        const auto model = dynamic_cast<MODEL *>(core.GetEntityPointer(headModel_));
+
+        static CMatrix mtx;
+        mtx.BuildPosition(0.f, 0.025f, 0.f);
+
+        static CMatrix mtx2;
+        mtx2.m[0][0] = 1.0f;
+        mtx2.m[1][1] = 1.0f;
+        mtx2.m[2][2] = 1.0f;
+
+        static CMatrix mtx3;
+        mtx3.BuildMatrix(0.1f, PI - 0.1f, 0.0f);
+
+        static CMatrix mtx4;
+        mtx4.BuildPosition(0.f, 0.f, 4.f);
+
+        model->mtx = mtx * mtx2 * mtx3 * mtx4;
+
+        model->GetAnimation()->CopyPlayerState(0, 1);
+        model->GetAnimation()->Player(0).SetAction("dialog_all");
+        model->GetAnimation()->Player(0).Play();
+        model->GetAnimation()->Player(0).SetAutoStop(false);
+
+        gs->SetTexturePath("");
+    }
+}
+
+void LegacyDialog::DrawHeadModel(uint32_t deltaTime)
+{
+    if (headModel_)
+    {
+        D3DVIEWPORT9 viewport;
+        RenderService->GetViewport(&viewport);
+
+        CMatrix mtx, view, prj;
+        uint32_t lightingState, zenableState;
+        RenderService->GetTransform(D3DTS_VIEW, view);
+        RenderService->GetTransform(D3DTS_PROJECTION, prj);
+        RenderService->GetRenderState(D3DRS_LIGHTING, &lightingState);
+        RenderService->GetRenderState(D3DRS_ZENABLE, &zenableState);
+
+        mtx.BuildViewMatrix(CVECTOR(0.0f, 0.0f, 0.0f), CVECTOR(0.0f, 0.0f, 1.0f), CVECTOR(0.0f, 1.0f, 0.0f));
+        RenderService->SetTransform(D3DTS_VIEW, (D3DXMATRIX *)&mtx);
```

Comment:
Please, use `(D3DMATRIX *)` instead of `(D3DXMATRIX *)`

Reply:
Updated it